Server 329 script to install puppet on sut

jpinsonault opened the pull request.

nwolfe commented on jenkins-integration/beaker/install/foss/40_add_puppet_bin_to_path.rb:
```diff
@@ -0,0 +1,6 @@
+
+hosts.each do |host|
+  on(host, 'echo "export PATH=\$PATH:/opt/puppetlabs/puppet/bin" >> ~/.bashrc')
+  on(host, 'echo "export PATH=\$PATH:/opt/puppetlabs/server/bin" >> ~/.bashrc')
```
Do we symlink all the binaries into `/opt/puppetlabs/bin/`?  If so we could just add that path

jpinsonault replied:
Yes, we do. Good call

nwolfe commented on jenkins-integration/scripts/foss_install.sh:
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
```
This file should be executable (the `pe_install.sh` is)

nwolfe commented on jenkins-integration/scripts/foss_install.sh:
```diff
@@ -0,0 +1,17 @@
+  --root-keys                  \
+  --repo-proxy                 \
+  --debug                      \
+  --keyfile $SSH_KEYFILE
```
https://github.com/puppetlabs/puppet-server/blob/master/acceptance/scripts/generic/testrun-full.sh

This is a good pattern to follow for writing these scripts. We should do the `--keyfile` part here and hardcode it

jpinsonault replied:
I like it, I'll convert it over

erikPrime replied:
We need some additional work here:
1.  We need to pass the name of an existing SUT in to the script.  Something like "perf-bl15"
2.  We need plug that host name into the host.yaml file, or into beaker somewhere.
3.  We need to pass --no-provision

harder stuff.
I think we should CONSIDER using a git submodule to pull in the helper.rb instead of using copy pasta.  BUT, maybe not because git submodule can be scary bad.

nwolfe replied:
@erikPrime Good points.  These scripts are far from done, and we're re-building the automation around them.  I imagine we will continue to generalize them as necessary so they can be called by other scripts and various information passed in.

For now, we're running them by hand, but soon will be writing the scripts that configure all the various variables and then call these.

> I think we should CONSIDER using a git submodule to pull in the helper.rb instead of using copy pasta.

Our `helper.rb` will be unique - despite the name being the same in all the projects, they don't actually contain the same content.  I believe the naming is allude to the fact that it's the file you pass to the Beaker `--helper` argument.

jpinsonault commented:
@erikPrime @nwolfe 
Addressed your feedback,
Updated gitignore for bundler
Converted beaker running scripts to a nicer format
Assume key file will be ~/.ssh/id_rsa-acceptance
Made foss_install.sh executable
Add /opt/puppetlabs/bin to path
Used cleaner on(hosts) instead of foreach loop where possible

Looking at @erikPrime's latest comment now

nwolfe commented on jenkins-integration/beaker/install/foss/40_add_puppet_bin_to_path.rb:
```diff
@@ -0,0 +1,5 @@
+
+# hosts.each do |host|
+  on(hosts, 'echo "export PATH=\$PATH:/opt/puppetlabs/bin" >> ~/.bashrc')
+# end
```
We should get rid of this file if it's only meant for us humans if/when we log in to the system :smile:

nwolfe commented on jenkins-integration/scripts/foss_install.sh:
```diff
@@ -0,0 +1,24 @@
+export BEAKER_CONFIG="scripts/hosts.cfg"
+export BEAKER_KEYFILE="~/.ssh/id_rsa-acceptance"
+export BEAKER_HELPER="beaker/helper.rb"
+export BEAKER_INSTALL_TYPE="foss"
```
Hmm...I'm wondering if this is the correct value to use here.  I'm seeing `aio` used now.  Not sure if we should change this though, but if we run into pathing problems we should keep this in mind.

jpinsonault replied:
That's a good question, I'm not sure what the difference is either

jpinsonault commented:
@nwolfe Per our discussion I removed genconfig/sqa_utils
Also removed the script that adds the bin folder to the path.

nwolfe commented on jenkins-integration/scripts/foss_install.sh:
```diff
@@ -0,0 +1,19 @@
+export BEAKER_TESTSUITE="${BEAKER_TESTSUITE:-beaker/install/foss/}"
+export BEAKER_KEYFILE="~/.ssh/id_rsa-acceptance"
+export BEAKER_HELPER="beaker/helper.rb"
+export BEAKER_INSTALL_TYPE="foss"
```
This Beaker `--type` argument determines file paths for a bunch of things, which in the past have been different between FOSS and PE installations.  In the new world, the paths for FOSS and PE have been unified under new locations, which are available with the `aio` argument.

I think this means we can always use `--type aio` for these scripts, regardless of FOSS or PE.  This would only break things if we try running against pre-AIO stuff, but I'm not sure that we're going to.

@jpinsonault If we run this with `"aio"` and it works, then I think we should use that.  I'll test this out in a minute.

jpinsonault replied:
@nwolfe Ahh, ok, thanks for the info
Tested it out on my end and it works fine with aio

jpinsonault commented:
@nwolfe Went ahead and switched the install type to aio, since it works fine in my testing

nwolfe commented on jenkins-integration/scripts/foss_install.sh:
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -x
+
+export GENCONFIG_LAYOUT="${GENCONFIG_LAYOUT:-centos6-64ma}"
```
Oops, guess we can get rid of this in these scripts.

jpinsonault replied:
D'oh! Thanks, pushed another commit

nwolfe commented on .gitignore:
```diff
@@ -13,3 +13,7 @@ cache
+
+# Bundler
+.bundle
+vendor
```
I noticed that beaker will generate a `log` directory - might as well add that too

jpinsonault replied:
Good call, I believe it creates a `junit` dir too, I'll add them both

jpinsonault commented:
Updated .gitignore with beaker output folders

nwolfe commented:
It works! :+1:

cprice404 commented on jenkins-integration/beaker/install/pe/10_install_pe.rb:
```diff
@@ -0,0 +1,3 @@
+test_name 'Install PE'
+
+install_pe
```
I presume this beaker method already has support for picking a specific version, configured via env vars or something?

jpinsonault replied:
Yeah, by environment variables or beaker host file config options. The readme in the `install` dir (in this PR) has some more info

cprice404 replied:
word

cprice404 commented:
I skimmed this; it looks good to me at a glance, and it seems like @nwolfe and @erikPrime have already been reviewing more thoroughly, so I'm +1 when they are!
